src/Skcore.jl

```diff
@@ -166,55 +166,66 @@ function import_sklearn()
     global mkl_checked
     global libstdcxx_solved
 
-    @static if Sys.isapple()
-      mod = try
+    mod = try
+        @static if Sys.isapple()
             if PyCall.conda && !mkl_checked
                 try
                     # check for existence of mkl-service. 
                     # Numpy, sklearn, etc. requires either `mkl` or `no-mkl` service to run
                     # By default Conda comes with mkl
-                    # For this package to run on MacOS the `no-mkl` versions of Numpy, sklearn is needed   
-                    pyimport("mkl")
+                    # For this package to run on MacOS, the `no-mkl` versions of Numpy, sklearn are needed   
+                    pyimport("mkl") # goes to catch block if "mkl" doesn't exist on the users Mac
                     
-                    #following Code runs only if mkl-service exists otherwise jumps to catch branch
+                    # following Code runs only if mkl-service exists otherwise jumps to catch branch
                     @info "Installing non-mkl versions of sci-kit learn via Conda"
-                    #use non-mkl versions of python packages when ENV["PYTHON"]="Conda" or "" is used
-                    #when a different non-conda local python is used everthing works fine
+                    # Use non-mkl versions of python packages when ENV["PYTHON"]="Conda" or "" is used
+                    # When a different non-conda local python is used everthing works fine
                     Conda.add("nomkl")
                     Conda.rm("mkl")#This also removes mkl-service
-                    #force reinstall of scikit-learn replacing any previous mkl version
-                    Conda.add("scikit-learn")
-                    Conda.add("openblas")
-                    Conda.add("llvm-openmp", channel = "conda-forge")
-                    mkl_checked = true
-                catch
-                    mkl_checked = true
-                end   
+                catch err
+                    ## This block is reached when `mkl` pkg isn't installed.
```

Comment:
I'm puzzled  as to why this catch block contains no code.

Reply:
Yeah, that's deliberate. I didn't wan't the try block to throw any error yet. I guess I should use a `try ... finally` there

Reply:
all good